### PR TITLE
Data Pipeline V2: Move Tabular Forecasting to the new API

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? [not needed for typos/docs]
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/api/tabular.rst [hunk 1]
```diff
@@ -41,10 +41,9 @@ ___________
 
     ~forecasting.model.TabularForecaster
     ~forecasting.data.TabularForecastingData
+    ~forecasting.input.TabularForecastingDataFrameInput
```
Comment (ethanwharris) [inline docs/source/api/tabular.rst:44]: This should probably be moved down and have the `~` removed

File: flash/tabular/__init__.py [hunk 2]
```diff
@@ -1,8 +1,4 @@
 from flash.tabular.classification import TabularClassificationData, TabularClassifier  # noqa: F401
 from flash.tabular.data import TabularData  # noqa: F401
-from flash.tabular.forecasting.data import (  # noqa: F401
-    TabularForecastingData,
-    TabularForecastingDataFrameInput,
-    TabularForecastingInputTransform,
-)
+from flash.tabular.forecasting.data import TabularForecastingData, TabularForecastingDataFrameInput  # noqa: F401
```
Comment (ethanwharris) [inline flash/tabular/__init__.py:3]: No need to import `TabularForecastingDataFrameInput` here:
```suggestion
from flash.tabular.forecasting.data import TabularForecastingData  # noqa: F401
```

**Review by ethanwharris — APPROVED** [review]
LGTM, small comment

File: docs/source/api/tabular.rst (hunk 1 above)
Comment (tchaton) [reply docs/source/api/tabular.rst:44]: Done !

**Pull request merged**